src/controllers/auth.controller.js

```
@@ -0,0 +1,149 @@
+import { User } from '../models/user.model.js';
+import { userService } from '../services/user.service.js';
+import { jwtService } from '../services/jwt.service.js';
+import { ApiError } from '../exeptions/api.error.js';
+import { tokenService } from '../services/token.service.js';
+import bcrypt from 'bcrypt';
+import {
+  validateEmail,
+  validateName,
+  validatePassword,
+} from '../utils/validation.js';
+
+const register = async (req, res) => {
+  const { name, email, password } = req.body;
+  const errors = {
+    name: validateName(email),
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+
+  if (name.email || errors.email || errors.password) {
+    throw ApiError.badRequest('Bad request', errors);
+  }
+
+  const hashedPass = await bcrypt.hash(password, 10);
+
+  await userService.register(name, email, hashedPass);
+  res.send({ message: 'OK' });
+};
+
+const activate = async (req, res) => {
+  const { activationToken } = req.params;
+  const user = await User.findOne({ where: { activationToken } });
+
+  if (!user) {
+    res.sendStatus(404);
+
+    return;
+  }
+  user.activationToken = null;
+  await user.save();
+
+  res.send(user);
+};
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+
+  const user = await userService.findByEmail(email);
+
+  if (!user) {
+    throw ApiError.badRequest('No such user');
+  }
+
+  const isPasswordValid = await bcrypt.compare(password, user.password);
+
+  if (!isPasswordValid) {
+    throw ApiError.badRequest('Wrong password');
+  }
+
+  await generateTokens(res, user);
+};
+
+const requestPasswordReset = async (req, res) => {
+  const { email } = req.body;
+
+  await userService.passwordReset(email);
```

Calling userService.passwordReset directly may leak whether an email exists depending on the service implementation. The password-reset request should not reveal user existence (avoid user enumeration). Ensure the service silently succeeds (e.g., always respond OK) or handle that behavior here. (Checklist: password reset request should not disclose existence).